modules/core/src/test/java/org/apache/ignite/internal/processors/cache/consistency/ReadRepairDataGenerator.java

```diff
@@ -218,6 +217,17 @@ public void generateAndCheck(
         }
     }
 
+    /**
+     * @param obj Object.
+     */
+    private Object describeArrayIfNeeded(Object obj) {
+        if (obj instanceof Object[])
+            return Arrays.deepToString((Object[])obj);
+        else if (obj instanceof int[])
```

Comment:
Let's user `obj != null && obj.getClass().isArray()` here. To handle all primitive arrays.

Reply:
You should call explicit method for each type. 
Each type has it's own `toString`.
Thats why only required impmented.